### PR TITLE
topn: coding style improved

### PR DESCRIPTION
**Pull request by jaroslavh**
Coding style improved for topn/topn.c

Compiles with warning, but works.

### CODE REVIEW

File: topn/topn.c [hunk 1]
```diff
@@ -45,380 +45,327 @@
...
 
-         case 'l':
-         interval = atoi(optarg);
-         if (interval == 0)
-         {
-            fprintf(stderr, "Invalid argument for parameter -l\n");
-            FREE_MODULE_INFO_STRUCT(MODULE_BASIC_INFO, MODULE_PARAMS);
-            TRAP_DEFAULT_FINALIZATION();
+      case 'l':
+           interval = atoi(optarg);
```
Comment (cejkato2) [inline topn/topn.c]: This seems to have a bad indentation - one more space?

File: topn/topn.c [hunk 2]
```diff
@@ -45,380 +45,327 @@
...
-   flow_t* record_bytes = malloc(sizeof(flow_t));
-   flow_t * array_of_packets = malloc(topn*sizeof(flow_t));
-   flow_t ** sorted_array_of_packets = malloc (topn*sizeof(flow_t*));
+   /* Counting Top N flows */
+   flow_t *array_of_bytes = malloc(topn * sizeof(flow_t));
+   flow_t **sorted_array_of_bytes = malloc (topn * sizeof(flow_t*));
+   flow_t *record_bytes = malloc(sizeof(flow_t));
+   flow_t *array_of_packets = malloc(topn * sizeof(flow_t));
+   flow_t **sorted_array_of_packets = malloc (topn * sizeof(flow_t*));
    flow_t* record_packets = malloc(sizeof(flow_t));
```
Comment (cejkato2) [inline topn/topn.c]: Replace "* " with " *"

File: topn/topn.c [hunk 3]
```diff
@@ -45,380 +45,327 @@
...
+
+   flow_t **array_of_bytes_port;
+   flow_t ***sorted_array_of_bytes_port;
+   flow_t **array_of_packets_port;
+   flow_t ***sorted_array_of_packets_port;
+   int *array_counter_port;
+
+   if (port_set != -1) {
+      array_of_bytes_port = malloc (port_cnt *(sizeof(flow_t *)));
+      sorted_array_of_bytes_port = malloc (port_cnt *(sizeof(flow_t **)));
```
Comment (cejkato2) [inline topn/topn.c]: There should be no space between malloc and '('

File: topn/topn.c [hunk 4]
```diff
@@ -436,30 +383,28 @@ int main(int argc, char **argv)
...
-      // Receive data from input interface (block until data are available)
+   while (!stop) {
+      /* Receive data from input interface (block until data are available) */
       ret = TRAP_RECEIVE(0, data, data_size, tmplt);
 
       TRAP_DEFAULT_RECV_ERROR_HANDLING(ret, continue, break);
 
-      // Check for end-of-stream message
+      /* Check for end-of-stream message */
       if (data_size <= 1)
```
Comment (cejkato2) [inline topn/topn.c]: Join the next line with '{'

File: topn/topn.c [hunk 5]
```diff
@@ -482,94 +427,74 @@ int main(int argc, char **argv)
...
+                     process_ip_pab(table_prefix_pab_port[p], &masked_ip, record_ip, data, tmplt,  average_packets / 3,  average_bytes / 3);
                   }
                }
             }
-
          }
-         else if ( prefix_only_v4 == -1 )
-         {
+         else if (prefix_only_v4 == -1) {
             masked_ip = (ur_get(tmplt, data, F_SRC_IP));
```
Comment (cejkato2) [inline topn/topn.c]: Join with the previous line:
`} else {`

File: topn/topn.c [hunk 6]
```diff
@@ -917,270 +799,212 @@ int main(int argc, char **argv)
...
-   {
+   if (num == 1) {
       if (sorted_array[0]->max_number > key) return -1;
       else return 0;
    }
 
-   if (num == 2)
-   {
+   if (num == 2) {
       if (sorted_array[0]->max_number > key) return -1;
```
Comment (cejkato2) [inline topn/topn.c]: Do not leave the statement of "true" branch of `if` without brackets, always use:
```
if () {
}
```
even for just one statement.

File: topn/topn.c [hunk 7]
```diff
@@ -917,270 +799,212 @@ int main(int argc, char **argv)
...
             if (middle == 0) return -1;
             middle--;
-         }    
+         }
          return middle;
-      }
-      else  last = middle - 1;
- 
-      middle = (first + last)/2;
+      } else  last = middle - 1;
```
Comment (cejkato2) [inline topn/topn.c]: The same for else statements, use brackets {}

File: topn/topn.c [hunk 8]
```diff
@@ -917,270 +799,212 @@ int main(int argc, char **argv)
...
-      printf("%d\t%s\t%s\t%d\t%d\t%d\t%u\n", y+1, ip_string, ip_string2, 
-      sorted_array_of_packets[i]->src_port,sorted_array_of_packets[i]->dst_port, sorted_array_of_packets[i]->protocol,sorted_array_of_packets[i]->max_number);
+      printf("%d\t%s\t%s\t%d\t%d\t%d\t%u\n", y + 1, ip_string, ip_string2,
+      sorted_array_of_packets[i]->src_port, sorted_array_of_packets[i]->dst_port, sorted_array_of_packets[i]->protocol, sorted_array_of_packets[i]->max_number);
    }
 }
 
-void print_top_ports(port_t * array_of_ports, int port_number)
-{
+void print_top_ports(port_t *array_of_ports, int port_number) {
```
Comment (cejkato2) [inline topn/topn.c]: For functions, '{' should be on the second line.

File: topn/topn.c [hunk 9]
```diff
@@ -917,270 +799,212 @@ int main(int argc, char **argv)
...
 
-   // --------------------------------- Flows
-
-   while (fht_get_next_iter(iter_flows) != FHF_ITER_RET_END) 
-   {
-      ip_t * zaznam = (void *) &iter_flows->data_ptr[0];
-      ip_addr_t * adresa = (void *) &iter_flows->key_ptr[0];
+   /* Flows */
+   while (fht_get_next_iter(iter_flows) != FHF_ITER_RET_END) {
+      ip_t *zaznam = (void *) &iter_flows->data_ptr[0];
```
Comment (cejkato2) [inline topn/topn.c]: Please, replace czech...

File: topn/topn.c [hunk 10]
```diff
@@ -1537,40 +1290,44 @@ void malloc_err(void)
...
 }
 
-int compare_flows (const void * a, const void * b)
+int compare_flows (const void *a, const void *b)
 {
-  if ( ((port_t*)a)->flows <  ((port_t*)b)->flows ) return 1;
-  if ( ((port_t*)a)->flows ==  ((port_t*)b)->flows ) return 0;
-  if ( ((port_t*)a)->flows >  ((port_t*)b)->flows ) return -1;
+  if (((port_t *) a)->flows < ((port_t *) b)->flows) return 1;
+  if (((port_t*) a)->flows == ((port_t*) b)->flows) return 0;
```
Comment (cejkato2) [inline topn/topn.c]: Put a space between type and * in casting to pointer.

**Comment by jaroslavh** [comment]
Warnings included in the following file:
[make_warnings.txt](https://github.com/CESNET/Nemea-Modules/files/750753/make_warnings.txt)

Most of them is printing out data from dynamically allocated memory where it is not checked whether the memory was rewriten after allocating it.

Also few functions with return type int have 3 if statements in them and no default return value.

To repair this I would have to go a little bit deeper into the program, tell me if I should do it.

**Pull request merged**